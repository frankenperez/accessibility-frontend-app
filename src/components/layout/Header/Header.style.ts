import styled from "@emotion/styled";

const StyledHeader = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 60px;
  padding: 10px 26px;
  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.16);

  .logo {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    gap: 12px;
  }

  .user {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    gap: 12px;

    .avatar {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 36px;
      height: 36px;
      background-color: ${({ theme }) => theme.palette.grey[100]};
      color: ${({ theme }) => theme.palette.primary.main};
      font-size: 16px;
      font-weight: 500;
      text-transform: uppercase;
    }

    .username {
      font-size: 1rem;
      color: ${({ theme }) => theme.palette.text.secondary};
    }
  }
`;

export default StyledHeader;
