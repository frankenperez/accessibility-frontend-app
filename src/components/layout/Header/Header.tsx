// React
import React from "react";

// Components
import { Avatar as MuiAvatar, IconButton as MuiIconButton } from "@mui/material";
import LogoutRoundedIcon from "@mui/icons-material/LogoutRounded";

// i18n
import { useTranslation } from "react-i18next";

// Styles
import StyledHeader from "./Header.style";

// Images
import logo from "assets/a11y.svg";

// Footer
export default function Header(): JSX.Element {
  const { t } = useTranslation("app");

  const handleLogout = () => {
    console.log("Logout");
  };

  function stringAvatar(name: string, surname: string) {
    return {
      children: `${name[0]}${surname[0]}`
    };
  }

  // TODO: Get Account details to show user
  return (
    <StyledHeader>
      <div className="logo">
        <img src={logo} alt="Accessibility Logo" />
        <span>Web Accessibility Project</span>
      </div>
      <div className="user">
        <MuiAvatar className="avatar" {...stringAvatar("Name", "Surname")} />
        <span className="username">Name Surname</span>
        <MuiIconButton aria-label={t("action.logout")} onClick={handleLogout} size="small">
          <LogoutRoundedIcon />
        </MuiIconButton>
      </div>
    </StyledHeader>
  );
}
