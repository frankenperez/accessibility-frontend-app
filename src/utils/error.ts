import axios from "axios";

export const errorParser = (error: unknown) => {
  let message: string;
  if (axios.isAxiosError(error)) {
    const errorData = error.response?.data as { reason: string } | string;
    if (typeof errorData === "object" && errorData?.reason) {
      message = errorData?.reason;
    } else {
      message = JSON.stringify(errorData);
    }
  } else if (error instanceof Error) {
    message = error.message;
  } else {
    message = JSON.stringify(error);
  }
  return message;
};
