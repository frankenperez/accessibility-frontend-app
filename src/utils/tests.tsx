// React
import React, { PropsWithChildren } from "react";

// Testing Library
import { render, RenderOptions } from "@testing-library/react";

// Router
import { BrowserRouter } from "react-router-dom";

// Internationalization
import en_BG from "../languages/en-GB.json";
import i18next from "i18next";
import { I18nextProvider } from "react-i18next";

// MUI Theme
import { createTheme, StyledEngineProvider, ThemeProvider } from "@mui/material/styles";
import theme from "styles/theme";
const appTheme = createTheme(theme);

// Internationalization config
i18next.init({
  interpolation: { escapeValue: false }, // React already does escaping
  lng: "en", // Language to use
  resources: {
    en: {
      app: en_BG
    }
  }
});

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  route?: string;
}

const renderWithProviders = (
  ui: React.ReactElement,
  {
    route = "/",
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren): JSX.Element => {
    return (
      <I18nextProvider i18n={i18next}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={appTheme}>
            <BrowserRouter>{children}</BrowserRouter>
          </ThemeProvider>
        </StyledEngineProvider>
      </I18nextProvider>
    );
  };
  window.history.pushState({}, "Test page", route);

  // Return an object with the store and all of RTL's query functions
  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders };
