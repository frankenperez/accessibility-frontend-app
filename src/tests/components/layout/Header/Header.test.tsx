// React
import React from "react";

// Component
import Header from "components/layout/Header/Header";

// Test Utils
import { renderWithProviders } from "utils/tests";

// Tests
describe("Tests for <Header />", () => {
  it("Renders with default props", () => {
    const { container } = renderWithProviders(<Header />);
    // Creates the snapshot test
    expect(container).toMatchSnapshot();
  });
});
