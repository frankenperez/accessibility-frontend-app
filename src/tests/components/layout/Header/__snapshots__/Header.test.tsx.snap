// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Header /> Renders with default props 1`] = `
<div>
  <header
    class="css-vrfcne"
  >
    <div
      class="logo"
    >
      <img
        alt="Accessibility Logo"
        src="a11y.svg"
      />
      <span>
        Web Accessibility Project
      </span>
    </div>
    <div
      class="user"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-r1updj-MuiAvatar-root"
      >
        NS
      </div>
      <span
        class="username"
      >
        Name Surname
      </span>
      <button
        aria-label="action.logout"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1gp9nik-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
          data-testid="LogoutRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 5h6c.55 0 1-.45 1-1s-.45-1-1-1H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h6c.55 0 1-.45 1-1s-.45-1-1-1H5V5z"
          />
          <path
            d="m20.65 11.65-2.79-2.79c-.32-.32-.86-.1-.86.35V11h-7c-.55 0-1 .45-1 1s.45 1 1 1h7v1.79c0 .45.54.67.85.35l2.79-2.79c.2-.19.2-.51.01-.7z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
